sunbeam-python/sunbeam/steps/microovn.py

```diff
@@ -0,0 +1,306 @@
+# SPDX-FileCopyrightText: 2025 - Canonical Ltd
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import logging
+import traceback
+import typing
+
+from rich.status import Status
+
+from sunbeam.clusterd.client import Client
+from sunbeam.clusterd.service import (
+    ConfigItemNotFoundException,
+    NodeNotExistInClusterException,
+)
+from sunbeam.core.common import (
+    BaseStep,
+    Result,
+    ResultType,
+    Role,
+    read_config,
+    update_config,
+)
+from sunbeam.core.deployment import Deployment
+from sunbeam.core.juju import (
+    ActionFailedException,
+    ApplicationNotFoundException,
+    JujuHelper,
+    JujuStepHelper,
+)
+from sunbeam.core.manifest import Manifest
+from sunbeam.core.openstack import OPENSTACK_MODEL
+from sunbeam.core.openstack_api import remove_hypervisor
+from sunbeam.core.steps import DeployMachineApplicationStep
+from sunbeam.core.terraform import TerraformHelper
+from sunbeam.lazy import LazyImport
+
+if typing.TYPE_CHECKING:
+    import openstack
+else:
+    openstack = LazyImport("openstack")
+
+LOG = logging.getLogger(__name__)
+CONFIG_KEY = "TerraformVarsMicroovnPlan"
+CONFIG_DISKS_KEY = "TerraformVarsMicroovn"
+APPLICATION = "microovn"
+MICROOVN_APP_TIMEOUT = 1200
+MICROOVN_UNIT_TIMEOUT = 1200
+
+
+class DeployMicroOVNApplicationStep(DeployMachineApplicationStep):
+    """Deploy MicroOVN application using Terraform."""
+
+    def __init__(
+        self,
+        deployment: Deployment,
+        client: Client,
+        tfhelper: TerraformHelper,
+        jhelper: JujuHelper,
+        manifest: Manifest,
+        model: str,
+    ):
+        super().__init__(
+            deployment,
+            client,
+            tfhelper,
+            jhelper,
+            manifest,
+            CONFIG_KEY,
+            APPLICATION,
+            model,
+            [Role.NETWORK],
+            "Deploy OpenStack microovn",
+            "Deploying OpenStack microovn",
+        )
+        self.openstack_model = OPENSTACK_MODEL
+
+    def get_application_timeout(self) -> int:
+        """Return application timeout in seconds."""
+        return MICROOVN_APP_TIMEOUT
+
+    def extra_tfvars(self) -> dict:
```

Comment:
Can you define channels in https://github.com/canonical/snap-openstack/blob/main/sunbeam-python/sunbeam/versions.py and use them to pass as tfvars